jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/traveler/TimController.java

```diff
@@ -72,87 +89,87 @@ public TimController(OdeProperties odeProperties) {
    @ResponseBody
    @CrossOrigin
    @RequestMapping(value = "/tim/query", method = RequestMethod.POST)
-   public ResponseEntity<String> queryForTims(@RequestBody String jsonString) { // NOSONAR
+   public ResponseEntity<String> asyncQueryForTims(@RequestBody String jsonString) { // NOSONAR
 
-      if (null == jsonString) {
+      if (null == jsonString || jsonString.isEmpty()) {
          logger.error("Empty request.");
-         return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Empty request");
+         return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(jsonKeyValue(ERRSTR, "Empty request."));
       }
 
-      HashMap<Integer, Boolean> resultTable = new HashMap<>();
+      ConcurrentHashMap<Integer, Integer> resultTable = new ConcurrentHashMap<>();
       RSU queryTarget = (RSU) JsonUtils.fromJson(jsonString, RSU.class);
 
-      SnmpSession ss = null;
+      SnmpSession snmpSession = null;
       try {
-         ss = new SnmpSession(queryTarget);
+         snmpSession = new SnmpSession(queryTarget);
+         snmpSession.startListen();
       } catch (IOException e) {
-         logger.error("Error creating TIM query SNMP session", e);
-         return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
-      } catch (NullPointerException e) {
-         logger.error("TIM query error, malformed JSON", e);
-         return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Malformed JSON");
+         logger.error("Error creating SNMP session.", e);
+         return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+               .body(jsonKeyValue(ERRSTR, "Failed to create SNMP session."));
       }
 
       // Repeatedly query the RSU to establish set rows
+      ExecutorService es = Executors.newFixedThreadPool(odeProperties.getRsuSrmSlots());
```

Comment:
A thread pool is good only when it creates the threads once and uses them (check-in/check-out) as needed. In this case, the pool is being created on every TIM query request which defeats the purpose of having a pool. I think you want the pool (es and give it a ore descriptive name like `threadPool`) be `static` with enough threads in there to support as many concurrent requests as needed. Not that we need thousands of threads but at least 2-3 times the number of RSU slots should be enough. So for example if we have say 300 threads in the pool and 5 request come in at the same time, 3 of then get serviced right away and the other two may have to wait until more threads are freed up.